syntax = "proto3";
package gateway;
option go_package = "usov.gateway.v1;gateway";



message register_request {
  string email = 1;
  string password = 2;
  string name = 3;
  string surname = 4;
}

message register_response {
  int64 user_id = 1;
}



message login_request {
  string email = 1;
  string password = 2;
  int64 app_id = 3;
}

message login_response {
  string token = 1;
}



message is_admin_request {
  int64 user_id = 1;
}

message is_admin_response {
  bool is_admin = 1;
}





service ExternalApiGateway {
  rpc register (register_request) returns (register_response);
  rpc login (login_request) returns (login_response);
  rpc is_admin (is_admin_request) returns (is_admin_response);
}